solr/core/src/test/org/apache/solr/core/TestCoreDiscovery.java

```diff
@@ -328,7 +328,7 @@ public void testTooManyTransientCores() throws Exception {
       // Just check that the proper number of cores are loaded since making the test depend on order
       // would be fragile
       assertEquals(
-          "There should only be 3 cores loaded, coreLOS and two coreT? cores",
+          "There should only be 3 cores loaded, coreLOS and two coreT? cores. Actual: "+cc.getLoadedCoreNames(),
```

Comment:
The code at https://github.com/apache/solr/blob/0c73f3e64f46e8ae1134f2f19f8f0ad29990c9e5/solr/core/src/java/org/apache/solr/core/TransientSolrCoreCacheDefault.java#L94 suggests that 3 as an absolute limit here is tricky.

Could the test perhaps poll and accept "3 or 4 but eventually 3" values?